src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,121 @@
+package duke.ui;
+
+import java.util.List;
+
+import duke.tasks.Task;
+
+/**
+ * Represents the User Interface (UI) for the DailyTasks application.
+ * This class is responsible for displaying messages to the user
+ * and formatting task-related information.
+ */
+public class Ui {
+
+    /** The name of the bot. */
+    public static final String BOT_NAME = "DailyTasks";
+
+    /** Greeting message displayed to the user upon startup. */
+    public static final String GREETING = "Hello! I'm " + BOT_NAME + ", your awesome task planner!";
+
+    /**
+     * Formats a single task with its corresponding index for display.
+     *
+     * @param i The index of the task.
+     * @param task The task to be formatted.
+     * @return The formatted task as a string.
+     */
+    private static String formatSingleTask(int i, Task task) {
+        return i + "." + task.toString() + "\n";
+    }
+
+    /**
+     * Displays the welcome message to the user.
+     */
+    public String showWelcome() {
+        return Ui.formatOutputMessage(GREETING);
+    }
+
+    /**
+     * Formats a list of tasks for display.
+     *
+     * @param tasks The list of tasks to be displayed.
+     * @param filtered Whether the task list is filtered.
+     * @return The formatted task list as a string.
+     */
+    public String formatTaskListings(List<Task> tasks, boolean filtered) {
+        StringBuilder str = new StringBuilder();
+
+        str.append(filtered ? "Here are the filtered tasks:" : "Here are the tasks in your list:")
+                .append("\n");
+        for (int i = 0; i < tasks.size(); i++) {
+            str.append(Ui.formatSingleTask(i + 1, tasks.get(i)));
+        }
+
+        return str.toString();
+    }
+
+    /**
+     * Formats an output message to be displayed within borders.
+     *
+     * @param input The message to be displayed.
+     * @return The formatted message as a string.
+     */
+    public static String formatOutputMessage(String input) {
+        return input;
+    }
+
+    /**
+     * Formats a message when a task is added to the list.
+     *
+     * @param taskCount The number of tasks in the list.
+     * @param task The task that was added.
+     * @return The formatted message as a string.
+     */
+    public static String formatAddTask(int taskCount, Task task) {
+        return (
+                "Got it. I've added this task:" + "\n" + task.toString() + "\n"
+                        + "Now you have " + taskCount + " tasks in the list." + "\n"
+            );
+    }
```

Comment:
Not sure why you decided to wrap the return statement this way, but it is still highly readable, which is great!